### ISSUE TITLE
support title and description in argparse add_mutually_exclusive_group

### ISSUE THREAD

**Issue by cjerdonek** [python/cpython#61420]
BPO | [17218](https://bugs.python.org/issue17218)
--- | :---
Nosy | @warsaw, @bitdancer, @cjerdonek
Files | <li>[test_argparse_mutex_with_title.py](https://bugs.python.org/file34845/test_argparse_mutex_with_title.py "Uploaded as text/plain at 2014-04-14.20:28:56 by tati_alchueyr"): proposed behavior for add_mutually_exclusive_group() when title and description are provided</li><li>[test_argparse_mutex_with_title1.py](https://bugs.python.org/file34926/test_argparse_mutex_with_title1.py "Uploaded as text/plain at 2014-04-16.20:03:49 by paul.j3")</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2013-02-17.00:38:22.020>
labels = ['type-feature', 'library']
title = 'support title and description in argparse add_mutually_exclusive_group'
updated_at = <Date 2016-09-20.23:44:11.933>
user = 'https://github.com/cjerdonek'
```

bugs.python.org fields:
```python
activity = <Date 2016-09-20.23:44:11.933>
actor = 'barry'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Library (Lib)']
creation = <Date 2013-02-17.00:38:22.020>
creator = 'chris.jerdonek'
dependencies = []
files = ['34845', '34926']
hgrepos = []
issue_num = 17218
keywords = []
message_count = 8.0
messages = ['182251', '216193', '216203', '216505', '216559', '216574', '216761', '216763']
nosy_count = 7.0
nosy_names = ['barry', 'bethard', 'r.david.murray', 'chris.jerdonek', 'tshepang', 'paul.j3', 'tati_alchueyr']
pr_nums = []
priority = 'normal'
resolution = None
stage = 'test needed'
status = 'open'
superseder = None
type = 'enhancement'
url = 'https://bugs.python.org/issue17218'
versions = ['Python 3.4']
```

</p></details>

**Comment by serhiy-storchaka** [comment]
`add_mutually_exclusive_group` on `add_argument_group` implement two orthogonal features. They share implementation, but this is an implementation detail. Perhaps #87425 would be more general solution to this issue.

**Comment by serhiy-storchaka** [comment]
Actually, adding mutually exclusive groups on argument groups always was possible. The only problem was with inherited parsers -- nested mutually exclusive groups lose their nesting after inheriting. But this should be fixed in #61011.

Note also that nesting argument groups in argument groups and mutually exclusive groups in mutually exclusive groups was deprecated. The fact that they currently share implementation is just an implementation detail, and in future this may be changed.

Closing this issue as mutually exclusive groups and argument groups are orthogonal features and creating a mutually exclusive group in a argument groups already gives you the desired result. Let not complicate the API.
